src/hooks/useL1CrossDomainMessengerProxy.ts

```diff
@@ -130,43 +109,42 @@ export function useL1CrossDomainMessengerProxy({
   const contract = getContract('L1CrossDomainMessengerProxy', fromChain.id)
   const { writeContractAsync } = useWriteContract()
 
-  const { gas: l2Gas, message } = useL2ContractCallInfo({
+  return useCallback(async () => {
+    const { gas: l2Gas, message } = await l2ContractCallInfo({
+      contractName,
+      functionName,
+      args,
+      value,
+      walletAddress,
+      chain: fromChain == sepolia ? optimismSepolia : optimism,
+    })
+
+    const l1Gas = await estimateGasL1CrossDomainMessenger({
+      chain: fromChain,
+      message,
+      value,
+      l2Gas,
+    })
+
+    return writeContractAsync({
```

Comment:
according to the doc, it's recommended to use `write` along with `simulate`, it would be fantastic if we can implement those good practices in the codebase: https://wagmi.sh/core/api/actions/writeContract#usage